### PR TITLE
feat: error handling

### PR DESCRIPTION
**Pull request by anusha-kommu**
Screenshot 2023-02-22 at 11 04 03

Screenshot 2023-02-22 at 11 01 56

Screenshot 2023-02-22 at 12 47 34

Screenshot 2023-02-20 at 12 49 56

Screenshot 2023-02-22 at 12 50 07

### CODE REVIEW

File: generator/overrides/education/pages/issuer/credential-form/index.tsx [hunk 1]
```diff
@@ -1,16 +1,25 @@
-import { FC } from 'react'
+import { FC, useEffect } from 'react'
 import { Formik } from 'formik'
 
 import { JSONLD_CONTEXT_URL } from 'utils/schema'
+import { messages } from 'utils/messages'
 import { useAuthContext } from 'hooks/useAuthContext'
 import { Container, Header, Input, Spinner } from 'components'
 
+import { notifyError } from '../../../utils/notification'
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/pages/issuer/credential-form/index.tsx:9]:
```suggestion
import { notifyError } from 'utils/notification'
```

File: generator/overrides/education/utils/notification.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { toast } from "components/Toast/Toast"
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/utils/notification.ts:1]:
```suggestion
import { toast } from "components"
```

File: generator/overrides/education/utils/notification.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+export const notifyError = (error: any) => {
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/utils/notification.ts:3]:
```suggestion
export const showErrorToast = (error: any) => {
```

File: generator/template/assets/info-filled-icon.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { SVGAttributes } from 'react'
+
+export const InfoFilledIcon = (props: SVGAttributes<SVGElement>) => (
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (OleksiiO-affinidi) [inline generator/template/assets/info-filled-icon.tsx:4]: this related to all icons
```suggestion
<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
```

File: generator/overrides/ticketing/pages/issuer/credential-form/index.tsx [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
 import * as S from './CredentialForm.styled'
 
-
 const CredentialForm: FC = () => {
-  const { handleSubmit, validate, isCreating } = useCredentialForm()
+  const { handleSubmit, validate, isCreating, error } = useCredentialForm()
+
+  useEffect(() => {
+    if (error) {
+      notifyError(new Error(messages.issuer.error.apiError))
```
Comment (OleksiiO-affinidi) [inline generator/overrides/ticketing/pages/issuer/credential-form/index.tsx:18]: didn't change
and do not forget to add it as dependency to `useEffect`

File: generator/overrides/ticketing/pages/issuer/credential-form/index.tsx [hunk 6]
```diff
@@ -1,15 +1,23 @@
-import { FC } from 'react'
+import { FC, useEffect } from 'react'
 import { Formik } from 'formik'
 
 import { Container, Header, Input, Textarea } from 'components'
 import { JSONLD_CONTEXT_URL } from 'utils/schema'
+import { messages } from 'utils/messages'
+
```
Comment (OleksiiO-affinidi) [inline generator/overrides/ticketing/pages/issuer/credential-form/index.tsx:7]: remove an empty line

File: generator/template/pages/holder/confirm-sign-in/useHolderConfirmSignIn.tsx [hunk 7]
```diff
@@ -12,16 +12,24 @@ export const useHolderConfirmSignIn = () => {
...
   const { authState, updateAuthState } = useAuthContext()
-  const { data, error, mutate, isLoading } = useConfirmSignInMutation()
-  const { data: signInData, mutateAsync: signInMutateAsync } = useSignInMutation()
-  const { computedCode, inputs, isButtonDisabled } = useConfirmSignIn(error?.message)
+  const { data, error, mutate, isLoading, reset } = useConfirmSignInMutation()
+  const { data: signInData, mutate: signInMutate } = useSignInMutation()
+  const { computedCode, inputs, isButtonDisabled, resetInputs } = useConfirmSignIn(error)
+
+  useEffect(() => {
+    if (error && computedCode.length < 6) {
```
Comment (OleksiiO-affinidi) [inline generator/template/pages/holder/confirm-sign-in/useHolderConfirmSignIn.tsx:20]: can u replace `6` with `inputs.length` pls

File: generator/overrides/education/components/Input/Input.styled.tsx [hunk 8]
```diff
@@ -76,6 +76,7 @@ export const Input = styled.input<Pick<Props, '$hasError'> & { $hasIcon: boolean
   transition: border-color 0.125s ease-in-out;
   outline: none;
   border: 1px solid ${(props) => props.theme.colors.neutral.primary['15']};
+  border-color: ${(props) => props.$hasError && props.theme.colors.utility.danger['100']};
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/components/Input/Input.styled.tsx:79]: this component already has `hasError` section 
pls use it instead

File: generator/overrides/ticketing/pages/issuer/credential-form/index.tsx (hunk 5 above)
Comment (anusha-kommu) [reply generator/overrides/ticketing/pages/issuer/credential-form/index.tsx:18]: when added to useEffect, it says `unused dependency`

**Review by OleksiiO-affinidi — APPROVED** [review]

**Pull request merged**